openmc/lattice.py

```diff
@@ -414,6 +415,57 @@ def find(self, point):
                 return []
         return [(self, idx)] + u.find(p)
 
+    def clone(self, memo=None):
+        """Create a copy of this lattice with a new unique ID, and clones
+        all universes within this lattice.
+
+        Parameters
+        ----------
+        memo : dict or None
+            A nested dictionary of previously cloned objects. This parameter
+            is used internally and should not be specified by the user.
+
+        Returns
+        -------
+        clone : openmc.Lattice
+            The clone of this lattice
+
+        """
+
+        if memo is None:
+            memo = {}
+
+        # If no nemoize'd clone exists, instantiate one
+        if self not in memo:
+            clone = deepcopy(self)
+            clone.id = None
+
+            if self.outer is not None:
+                clone.outer = self.outer.clone(memo)
+
+            # Clone all unique universes in the lattice
+            univ_clones = self.get_unique_universes()
+            for univ_id in univ_clones:
+                univ_clones[univ_id] = univ_clones[univ_id].clone(memo)
```

Comment:
Come to think of it, why does this need to be done up front? You can just call `Universe.clone(memo)` below directly and it will be handled correctly.